bpo-35500: align expected and actual calls on mock.assert_called_with error message.

suhearsawho opened the pull request:
This patch adds an enhancement to the unittest module by adding a newline between the Assertion Error and Expected fields of the error messages generated by mock.assert_called_with. With this addition, the output is clearer as it is easier to read. 

<!-- issue-number: [bpo-35500](https://bugs.python.org/issue35500) -->
https://bugs.python.org/issue35500
<!-- /issue-number -->

suhearsawho committed:
bpo-35500: align expected and actual calls on mock.assert_called_with error message

blurb-it[bot] committed:
📜🤖 Added by blurb_it.

taleinat commented on Misc/NEWS.d/next/Library/2019-02-10-00-00-13.bpo-35500.1HOMmo.rst:
```diff
@@ -0,0 +1 @@
+When mock.assert_called_with is called, the error message that is generated currently does not possess a newline between the AssertionError and the Expected field. Now, the message "expected call not found." will follow the AssertionError. In the next lines, the Expected and Actual values will be displayed.
```
Thanks for adding a NEWS entry!

This text would be a great description for the PR, but is too verbose for a NEWS entry. A NEWS entry should be a concise description of the change. In this case, I suggest something along the lines of:

```suggestion
Write expected and actual call parameters on separate lines in mock.assert_called_wtih() assertion errors.  Contributed by Susan Su.
```

taleinat committed:
Update Misc/NEWS.d/next/Library/2019-02-10-00-00-13.bpo-35500.1HOMmo.rst

Co-Authored-By: suhearsawho <susansu.software@gmail.com>

suhearsawho committed:
Update 2019-02-10-00-00-13.bpo-35500.1HOMmo.rst

taleinat reviewed (commented):
This looks good to me, but I'm still waiting for the issue creator to respond to the suggestions to replace the word "actual" with something longer to achieve vertical alignment between the "Expected" and "Actual" lines.

tirkarthi reviewed (approved):
LGTM as per https://bugs.python.org/issue35500#msg335145. Please sign the CLA.

Minor nit : You might want to use the below format for hyperlinking to `assert_called_with` method docs.

```rst
Write expected and actual call parameters on separate lines in :meth:`unittest.mock.Mock.assert_called_with` assertion errors.  Contributed by Susan Su.
```

Thanks @suhearsawho for your contribution.

suhearsawho committed:
Update 2019-02-10-00-00-13.bpo-35500.1HOMmo.rst

Changed format for hyperlinking to assert_called_with method (Suggestion from @tirkarthi).

taleinat reviewed (approved):
LGTM. Just waiting for approval of CLA signing.

suhearsawho committed:
bpo-35500: align expected and actual calls on mock.assert_called_with when self.call_args is None. This commit aims to create consistency with other case in mock.assert_called_with

suhearsawho committed:
Merge branch 'bpo-35500' of https://github.com/suhearsawho/cpython into bpo-35500
